### PR TITLE
Add support for global provider settings

### PR DESCRIPTION
**Pull request by danielpeach**
- Adds support for global provider settings
- Integrates with application-level provider settings (global settings are defaults on the create application modal).

@duftler @anotherchrisberry PTAL

### CODE REVIEW

File: app/scripts/modules/core/application/modal/applicationProviderFields.component.js [hunk 1]
```diff
@@ -14,8 +16,17 @@ module.exports = angular
       },
       controller: 'ApplicationProviderFieldsCtrl',
   })
-  .controller('ApplicationProviderFieldsCtrl', function(cloudProviderRegistry) {
+  .controller('ApplicationProviderFieldsCtrl', function(cloudProviderRegistry, settings, _) {
     const templateUrlPath = 'applicationProviderFields.templateUrl';
+    let defaultProviderFields = settings.providers;
+
+    this.initializeApplicationFields = (fieldPath) => {
```
Comment (anotherchrisberry) [inline app/scripts/modules/core/application/modal/applicationProviderFields.component.js]: `initializeApplicationFields` implies that multiple fields can be set by this method. It should either take a spread operator (`this.initializeApplicationFields = (...fieldPath) => `, then iterate over them) or be renamed to `initializeApplicationField`.

**Comment by anotherchrisberry** [comment]
Aside from the name of the method, LGTM - appreciate the included tests.

**Comment by danielpeach** [comment]
@anotherchrisberry I changed the method name.

**Comment by anotherchrisberry** [comment]
Thanks - LGTM.

**Pull request merged**